pandas/_libs/tslibs/period.pyi

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+
+from pandas._libs.tslibs.nattype import NaTType
+from pandas._libs.tslibs.offsets import BaseOffset
+from pandas._libs.tslibs.timestamps import Timestamp
+
+INVALID_FREQ_ERR_MSG: str
+DIFFERENT_FREQ: str
+
+class IncompatibleFrequency(ValueError): ...
+
+def periodarr_to_dt64arr(
+    periodarr: np.ndarray,  # const int64_t[:]
+    freq: int,
+) -> np.ndarray: ...  # np.ndarray[np.int64]
+
+def period_asfreq_arr(
+    arr: np.ndarray,  # ndarray[int64_t] arr,
+    freq1: int,
+    freq2: int,
+    end: bool,
+) -> np.ndarray: ... # np.ndarray[np.int64]
+
+def get_period_field_arr(
+    field: str,
+    arr: np.ndarray,  # const int64_t[:]
+    freq: int,
+) -> np.ndarray: ... # np.ndarray[np.int64]
+
+def from_ordinals(
+    values: np.ndarray,  # const int64_t[:]
+    freq,
```

Comment:
str, tuple, datetime.timedelta, DateOffset or None

same as type of freq to to_offset in pandas/_libs/tslibs/offsets.pyx, maybe need an alias since used in several places